src/content/protocols/compound-v3.md

```diff
@@ -0,0 +1,240 @@
+
+<!--
+
+Compound's upgrade process could potentially allow theft or loss of user funds under extreme circumstances. While upgrades require governance approval, if governance were compromised and the `ProposalGuardian` (security council) failed to cancel malicious proposals, attackers could theoretically deploy malicious contracts that steals all funds not withdrawn during the exit window.
+
+
+ The `Comet`s, `Configuration`, and `Compound Governor` contracts can be changed at any time with a delay of 2 days once a governance proposal is accepted. The `Comet`s are deployed by the `Comet Factory` using the `Configurator`. The `Configurator` holds the parameters of each market. The desired process to update the market is via the configurator that holds the possible market parameters and making the market (with all the funds of a respective base asset) point to this new logic. A `PauseGuardian` (security council) can pause the markets at any time and a `ProposalGuardian` can cancel goverance proposals
+before execution (incl. upgrades).
+
+When assuming that the governance could be hijacked and the `ProposalGuardian` acts in favor of this hijack (not stoping from enforcement of malicious proposal) then the current implementation does not prohibit that a market gets a completely malicious logic assigned that does not stem from the configurator itself and allows the attacker to steal all funds that were not rescued in the 2-day delay period. -->
+
+> Upgradeability score: High
+
+## Autonomy
+
+The system has one dependency. The protocol uses Chainlink's oracle to get the price of assets. There are no fallback mechanisms if the oracle fails. It may be replaced only with a contract upgrade through a governance proposal (5+ days delay).
+
+> Autonomy score: Low
+
+## Exit Window
+
+Once an upgrade is approved by the governance there is a delay of 2 days allowing users to react. Anyone with more than 25'000 Comp can create a proposal, each proposal has a minimum voting time of 3 days and requires at least 400'000 votes to be valid. A malicious upgrade could hijack user funds if it is not blocked by the `ProposalGuardian`.
+In addition to that, the tranfers/deposits/withdrawals can be paused by the `PauseGuardian` (Security Council) with no delay, freezing all assets.
+
+> Exit Window score: High
+
+## Accessibility
+
+The frontend of Compound V3 is open source. Instructions to deploy it locally or deploy it
+on IPFS are available [here](https://github.com/compound-finance/palisade). There is no registry
+of alternative deployments.
+
+> Accessibility score: Medium
+
+## Conclusion
+
+Deposited assets can be lost by malicious or faulty updates (low [upgradability](#upgradeability) score).
```

Comment:
The Compound-v3 Ethereum mainnet protocol achieves Low decentralization scores for its *Upgradeability*, *Autonomy* and *Exit Window* dimensions. It thus ranks Stage 0.

The protocol could reach Stage 1 by 1) adopting a *Security Council* setup for the `ProposalGuardian` and `PauseGuardian` multisig accounts, and 2) implementing validity checks and a fallback mechanism around the Chainlink oracle (or Chainlink adopting a *Security Council* setup for its own multisig account).